File: python/coding_exercise/exercise_1/code_1.py
def is_arrays_contain_common_item1(arr1: [str], arr2: [str]) -> bool:
    # If any array is empty return false
    if len(arr1) == 0 or len(arr2) == 0:
        return False

    # Find the biggest array list
    big_array = "arr2" if len(arr1) > len(arr2) else "arr1"

    # Convert the smallest array into the object
    big_array_obj = {}

    for item in (arr1 if big_array == "arr2" else arr2):
        big_array_obj[item] = True

    # Loop through the biggest array and check the item exists in the newly created object
    for item in (arr2 if big_array == "arr2" else arr1):
        if item in big_array_obj:
            return True

    return False

def is_arrays_contain_common_item2(arr1: [str], arr2: [str]) -> bool:
    return any(item in arr1 for item in arr2)

File: typescript/src/components/coding-exercise/arrays/exercise-1/code-1.ts
export const isArraysContainCommonItem1 = (
  arr1: string[],
  arr2: string[],
): boolean => {
  // If any array is empty return false
  if (arr1.length === 0 || arr2.length === 0) {
    return false;
  }

  // Find the biggest array list
  const bigArrayName: string = arr1.length > arr2.length ? "arr1" : "arr2";

  // Convert the smallest array into the object
  const arrObj: { [key: string]: boolean } = {};

  (bigArrayName === "arr1" ? arr2 : arr1).forEach((item) => {
    arrObj[item] = true;
  });

  // Loop through the biggest array and check the item exists in the newly created object
  for (let i = 0; i < (bigArrayName === "arr1" ? arr1 : arr2).length; i++) {
    if (arrObj[(bigArrayName === "arr1" ? arr1 : arr2)[i]]) {
      return true;
    }
  }

  return false;
};

export const isArraysContainCommonItem2 = (
  arr1: string[],
  arr2: string[],
): boolean => arr1.some((item) => arr2.includes(item));
